src/frontend/js/hooks/useUser.tsx

```diff
@@ -0,0 +1,31 @@
+import { defineMessages } from 'react-intl';
+import { API } from 'types/Joanie';
+import { useJoanieApi } from 'contexts/JoanieApiContext';
+import { JoanieUser } from 'types/User';
+import { QueryOptions, ResourcesQuery, useResource, UseResourcesProps } from './useResources';
+
+const messages = defineMessages({
+  errorGet: {
+    id: 'hooks.useUser.errorGet',
+    description: 'Error message shown to the user when user fetch request fails.',
+    defaultMessage: 'An error occurred while fetching logged user. Please retry later.',
+  },
+  errorNotFound: {
+    id: 'hooks.useUser.errorNotFound',
+    description: "Error message shown to the user when it isn't logged.",
+    defaultMessage: "You aren't log in.",
+  },
+});
+
+const props: UseResourcesProps<JoanieUser, ResourcesQuery, API['user']['me']> = {
+  queryKey: ['user'],
+  apiInterface: () => useJoanieApi().user.me,
+  session: true,
+  messages,
+};
+/**
+ * Joanie Api hook to retrieve the authenticated user.
+ */
+export const useUser = ({ enabled = true }: Pick<QueryOptions<JoanieUser>, 'enabled'> = {}) => {
+  return useResource(props)(undefined, {}, { enabled });
```

Comment:
I wonder if this is not easiest to use `useSessionQuery` directly for this specific query.

Reply:
Now that i moved it into `JoanieSessionProvider` i like to have this `useResource` to plug it the same way addresses, orders and creditCards are plugged.
The change on enable is no longer needed.